src/include/duckdb/parser/parsed_data/vacuum_info.hpp

```diff
@@ -9,11 +9,47 @@
 #pragma once
 
 #include "duckdb/parser/parsed_data/parse_info.hpp"
+#include "duckdb/parser/tableref.hpp"
+#include "duckdb/planner/tableref/bound_basetableref.hpp"
 
 namespace duckdb {
 
+enum class VacuumOptions : int { VACUUM = 1 << 0, ANALYZE = 1 << 1 };
+
 struct VacuumInfo : public ParseInfo {
-	// nothing for now
+public:
+	explicit VacuumInfo(const VacuumOptions &options) : options(options), has_table(false) {};
+
+	unique_ptr<VacuumInfo> Copy() {
+		auto result = make_unique<VacuumInfo>(options);
+		result->has_table = has_table;
+		if (has_table) {
+			result->ref = ref->Copy();
+		}
+		return result;
+	}
+
+	bool Vacuum() {
+		return HasOption(VacuumOptions::VACUUM);
+	}
+
+	bool Analyze() {
+		return HasOption(VacuumOptions::ANALYZE);
+	}
+
+public:
+	//! Bitmap of options set by the parser
+	const VacuumOptions options;
```

Comment:
The postgres parser returns a bitmap with all the options that are enabled, so it's not really an enum, since options can be combined.

I agree that a bitmap is not really needed. I could instead add a few `bool` fields to `VacuumOptions`, one for each option that we support. Would that be preferred?

Reply:
Yes, that sounds better/more clear